### ISSUE TITLE
Extract data from xgb.DMatrix object?

### ISSUE THREAD

**Issue by dguller01** [dmlc/xgboost#422]
I know we can put our training data into xgb.DMatrix object, but can we do the reverse?  I can extract the labels with the getinfo() function, but I can't figure out how to get the input columns.

Any help would be appreciated!

**Comment by tqchen** [comment]
Currently we cannot direct extract data matrix from DMatrix, mainly due to DMatrix is a internal data structure that may or maynot sit in memory.
